### ISSUE TITLE
What is the relationship with joyent/libuv

### ISSUE THREAD

**Issue by opoplawski** [JuliaAttic/libuv-archive#2]
What is the relationship with joyent/libuv?  Are you going to be submitting your changes there?  This will need to get resolved before julia can be properly packaged in Fedora.

**Comment by vtjnash** [comment]
We have been working with joyent to incorporate our changes upstream. There is an open pull request.

We are currently doing a static link against a private copy of libuv, to avoid dependency and version issues (so having libuv installed is not required for execution). We also make some assumptions in the code about the version of libuv, so it could be tricky to build against an arbitrary system version.

I was not aware that system package managers had started to include libuv, can you provide a link to the libuv in the Fedora package manager? Typically, package managers are not kept sufficiently up-to-date to handle such a tightly-coupled build dependency.

**Comment by opoplawski** [comment]
Glad to hear you are working together.

I'm currently playing with packaging julia (and hence libuv) for Fedora.  There is also an open review for node.js which will probably also require a separate libuv package.  Unless an exception is granted (which has been done before), this common libuv will need to serve both node.js and julia or any other packages in Fedora.

**Comment by Keno** [comment]
The problem is that the libuv API is not stable and probably never will (at least not in the near future) as it is regarded internal by node.js. In addition, there might be some libuv configuration options that will be different for julia and node.js (I'm thinking particularly integrating libuv more closely with Julia's type system, but I haven't really looked into that yet).

**Comment by vtjnash** [comment]
closing as invalid

**Comment by Keno** [comment]
Re-opened as requested by @sebastien-villemot in recognition of official upstream releases.

**Comment by nalimilan** [comment]
What's the progress regarding the merge of Julia patches upstream? In the long term, for distributions packages, Julia should really use the upstream libuv.

Do you have a link to the issues/PRs upstream?

**Comment by tkelman** [comment]
@nalimilan see
- https://github.com/joyent/libuv/issues?q=author%3Avtjnash
- https://github.com/joyent/libuv/issues?q=author%3Akeno
- https://github.com/joyent/libuv/issues?q=author%3Astaticfloat
- https://github.com/joyent/libuv/issues?q=author%3Aihnorton
- https://github.com/joyent/libuv/issues?q=author%3Atimholy

I think the biggest remaining local patch is the oldest one tracked in https://github.com/joyent/libuv/pull/451, and we haven't yet tried to upstream https://github.com/JuliaLang/libuv/pull/24 but perhaps should

**Comment by stevengj** [comment]
@Keno, regarding the lack of a stable libuv API, this seems to no longer be the case.  On https://github.com/libuv/libuv it says that they are now using semantic versioning to give some guarantees of ABI and API stability.

~~What are the remaining issue that block us from using the upstream libuv?~~ Oh, I see that @tkelman's post covers this.

**Comment by vtjnash** [comment]
The primary one I've been waiting on is https://github.com/libuv/libuv/issues/856. If they accept that, I think I can create a new version of joyent#451 that they will merge. Otherwise, I expect we'll announce a new name for our project fork and stop trying to upstream our bugfixes.

**Comment by PallHaraldsson** [comment]
libuv here, shows many commits ahead, and incorrectly one behind.

Is it possible to "link" it to the moved one, to show differences better?

At least this is of interest to me:

https://github.com/libuv/libuv/blob/v1.x/android-configure

Is support of libuv of Android, probably by now, the most obvious need to get Julia on Android supported?

**Comment by vtjnash** [comment]
We have that file too (https://github.com/JuliaLang/libuv/blob/julia-uv1.9.0/android-configure). Since libuv recreated their repo under a different account, we would need to destroy and re-fork ours to fix the link (or ask Github to do it for us).

**Comment by PallHaraldsson** [comment]
Sorry, I overlooked android-configure file here (assumed not here as not supported..), but e.g. that file is older.

Maybe, since [older] Android files are here, libuv isn't a hindrance (or could be updated). The kernel is the same.

Would be nice to have Github or someone "re-fork".. or even better merge with upstream. I can deal, I guess, do "diff" to look into needed Android support here (and also in general).

**Comment by tkelman** [comment]
our branch and the upstream branch currently diverge starting at https://github.com/JuliaLang/libuv/commit/229b3a4cc150aebd6561e6bd43076eafa7a03756

**Comment by stevengj** [comment]
@vtjnash, could you give a brief explanation of why this was closed?  What is the resolution on the relationship to what is now https://github.com/libuv/libuv?

**Comment by ararslan** [comment]
I can't speak to Jameson's reasoning for closing this, but I'll note that we do have some of our changes as PRs on https://github.com/libuv/libuv. PRs there are merged on geological timescales though, so I imagine we'll likely have to keep using our fork for Julia for the foreseeable future.
